bin/pgenv

```diff
@@ -260,20 +270,283 @@ pgenv_current_version_number(){
+    local file=$1
+    local name=$2
+    local value=$3
+    local comment=$4
+
+    if [ ! -z "comment" ]; then
+        echo "# $comment" >> "$file"
+    fi
+
+    # if no value supplied, put a comment
+    if [ -z "$value" ]; then
+        echo -n "# " >> "$file"
+    fi
+    echo "${name}='${value}'" >> "$file"
+    echo >> "$file"
+}
+
+# Writes the whole current configuration to the current version file
+# or the version file specified. It does a backup copy in the case
+# the file already exists.
+pgenv_configuration_write() {
+    local v=$1
+    CONF=$( pgenv_configuration_file_name $v )
+
+    # sanity check
+    if [ -z "$CONF" ]; then
+        echo "Cannot determine which configuration file to use!"
+        exit 1
+    fi
+
+    # make a backup copy
+    if [ -f "$CONF" ]; then
+        CONF_BACKUP="${CONF}.backup"
+        pgenv_debug "Making backup copy of [$CONF] into [$CONF_BACKUP]"
+        mv "$CONF" "$CONF_BACKUP"
+    fi
+
+    # ok, write the configuration
+    echo "# pgenv configuration created on $(date)" > "$CONF"
+    echo "" >> "$CONF"
+    pgenv_configuration_write_variable "$CONF" "PGENV_DEBUG" "" 'Enables debug output'
+    pgenv_configuration_write_variable "$CONF" "PGENV_STOP_MODE" "$PGENV_STOP_MODE" "Cluster stop mode for 'stop' and 'restart'"
+    pgenv_configuration_write_variable "$CONF" "PGENV_LOG" "$PGENV_LOG" "Path to the cluster log file"
+    pgenv_configuration_write_variable "$CONF" "PGENV_PG_USER" "$PGENV_PG_USER" "Cluster administrator user"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_INITDB" "$PGENV_FLAGS_INITDB" "Initdb flags"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_MAKE" "$PGENV_FLAGS_MAKE" "Make flags"
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_CONFIGURE" "$PGENV_FLAGS_CONFIGURE" "Configure flags"
+    pgenv_configuration_write_variable "$CONF" "PGENV_PLPERL" "$PGENV_PLPERL" 'Perl 5 executable to build PL/Perl'
+    pgenv_configuration_write_variable "$CONF" "PGENV_PLPYTHON" "$PGENV_PLpython" 'Python executable to build PL/Perl'
+    pgenv_configuration_write_variable "$CONF" "PGENV_MAKE" "$PGENV_MAKE" 'Make command to use for build'
+    pgenv_configuration_write_variable "$CONF" "PGENV_CURL" "$PGENV_CURL" 'Curl command to download source code'
+    pgenv_configuration_write_variable "$CONF" "PGENV_PATCH" "$PGENV_PATCH" 'Patch command for specific versions'
+    pgenv_configuration_write_variable "$CONF" "PGENV_SED" "$PGENV_SED" 'Sed used to manipulate strings'
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_PGCTL_START" "$PGENV_FLAGS_PGCTL_START" 'Start configuration flags'
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_PGCTL_stop" "$PGENV_FLAGS_PGCTL_STOP" 'Stop configuration flags'
+    pgenv_configuration_write_variable "$CONF" "PGENV_FLAGS_PGCTL_RESTART" "$PGENV_FLAGS_PGCTL_RESTART" 'Restart configuration flags'
+
+
+    echo "pgenv configuration written to file [$CONF]"
```

Comment:
No brackets, please.

Reply:
No problem, but I found them useful when (this is not the case) your filename could contain blanks.